pandas/core/indexing.py

```diff
@@ -933,7 +933,8 @@ def _getitem_tuple_same_dim(self, tup: tuple):
         This is only called after a failed call to _getitem_lowerdim.
         """
         retval = self.obj
-        for i, key in enumerate(tup):
+        for i, key in enumerate(reversed(tup)):
+            i = self.ndim - i - 1
```

Comment:
Comment about why doing it this way, otherwise future-me might try to “simplify” it

Reply:
Can also put this right before `retval = ...`

Reply:
Added a comment, prefer keeping it here. We might miss this if we use i before this at some point